autoload/airline/extensions/branch.vim

```diff
@@ -109,6 +110,49 @@ function! s:get_hg_branch()
   return ''
 endfunction
 
+if s:has_async
+  let s:jobs = {}
```

Comment:
The dictionary should be cleaned after the job is done, i.e. in `on_exit` call.

There's no need to keep that data record until the next job is started. In the current implementation, if the job for given file is never started again, then its record will never be cleaned.